random1on1/api/channels.py

```diff
@@ -1,36 +1,132 @@
+import json
 from abc import ABC
 from abc import abstractmethod
+from datetime import datetime
+from functools import reduce
 
-from discord.channel import GroupChannel
-from discord.channel import TextChannel
+from discord import CategoryChannel
+from discord import GroupChannel
+from discord import Role
+from discord import TextChannel
+from networkx import connected_components
+from networkx import union
+from preconditions import preconditions
 
-from random1on1.api.participant import Participant
+from .pairings import Pairings
 
 
-class ReadWriteChannel(ABC, TextChannel):
+class Random1on1Channel(ABC):
 
+    @preconditions(lambda name: len(name) > 0)
     @abstractmethod
-    def __init__(self, channel_name: str):
-        # TODO: Add precondition checks here
-        self.channel_name = channel_name
+    def __init__(self, name: str, category: CategoryChannel):
```

Comment:
`name` is the name of the channel which the bot will live in right?

`s/name/channel_name`

Reply:
Small nit, but I don't agree with this name change suggestion. It should be clear from context that the name refers to the channel name. 

In practice use it looks like 
```python
history_channel = HistoryChannel(name="history")
# ... 
assert history_channel.name == "history"
```
Adding redundant "channel_" just makes the code less readable (unless we choose at some point to add another name-field to the class in which case it's worth revisiting)

The convention of just `name` is also the same convention that the `discord.py` project uses.